### ISSUE TITLE
Executing cargo-process-run is different from cargo run -p xxx

### ISSUE THREAD

**Issue by mickvangelderen** [kwrooijen/cargo.el#62]
First of all, I'm not sure if this is a problem with cargo.el, but it is my best guess. I will try to explain my situation as best as possible. 

I'm working on an OpenGL project that I have been able to execute from emacs using cargo-process-run up until now. The project is in a workspace. After updating spacemacs to develop and all packages, I suddenly started getting the following error when running cargo-process-run:

```
thread 'main' panicked at 'Failed to initialize any backend!
    Wayland status: NoCompositorListening
    X11 status: XOpenDisplayFailed
```

When I run `cargo run -p xxx` from the command line, the application runs as it has before. 

Do you know of anything that has changed recently in emacs, rust-mode or one of its dependencies that might have changed the environment in which the process is ran?

**Comment by kwrooijen** [comment]
Hey, sorry to hear. It could be caused by recent changes to this package. Is your project open source by any chance so I can reproduce the issue? This error doesn't really tell me much sadly.

**Comment by mickvangelderen** [comment]
Yes it is: https://github.com/mickvangelderen/blocks-rust. Thank you for looking into this.

**Comment by mickvangelderen** [comment]
I'm on spacemacs development branch commit c122eb6a05ddffeccfe20fc7f56cae550c463575. Updated all my packages yesterday if that helps.

**Comment by kwrooijen** [comment]
Thanks for the info, I'll try to look into this soon

**Comment by mickvangelderen** [comment]
Yeah no worries. I really like being able to jump to the code from the cargo process output but for now I can get by with the terminal.
